ecommerce_integrations/woocommerce/woocommerce_connection.py

```diff
@@ -0,0 +1,273 @@
+		if not frappe.db.get_value(
+			"Ecommerce Item", {"erpnext_item_code": tmp_erpnext_item_code}, "name"
+		):
+			item_dict = {
+				"is_stock_item": 1,
+				"item_code": tmp_erpnext_item_code,
+				"item_name": cstr(item_data.get("name")),
+				"description": cstr(item_data.get("name")),
+				"item_group": PRODUCT_GROUP,
+				"has_variants": 0,
+				"woocommerce_id": item_woo_com_id,
+				"stock_uom": woocommerce_settings.uom or _("Nos", sys_lang),
+				"sku": item_data.get("sku"),
+			}
+			ecommerce_item.create_ecommerce_item(
+				integration=MODULE_NAME,
+				integration_item_code=item_data.get("sku"),
+				item_dict=item_dict,
+				sku=item_data.get("sku"),
+			)
+
+
+def create_sales_order(order, woocommerce_settings, customer_name, sys_lang):
+	new_sales_order = frappe.new_doc("Sales Order")
+	new_sales_order.customer = customer_name
+
+	new_sales_order.po_no = new_sales_order.woocommerce_id = order.get("id")
+	new_sales_order.naming_series = woocommerce_settings.sales_order_series or "SO-WOO-"
+
+	created_date = order.get("date_created").split("T")
+	new_sales_order.transaction_date = created_date[0]
+	delivery_after = woocommerce_settings.delivery_after_days or 7
+	new_sales_order.delivery_date = frappe.utils.add_days(created_date[0], delivery_after)
+
+	new_sales_order.company = woocommerce_settings.company
+
+	set_items_in_sales_order(new_sales_order, woocommerce_settings, order, sys_lang)
+	new_sales_order.flags.ignore_mandatory = True
+	new_sales_order.insert(ignore_permissions=True)
+	new_sales_order.submit()
+
+
+def set_items_in_sales_order(new_sales_order, woocommerce_settings, order, sys_lang):
+	company_abbr = frappe.db.get_value("Company", woocommerce_settings.company, "abbr")
+
+	default_warehouse = _("Stores - {0}", sys_lang).format(company_abbr)
+	if not frappe.db.exists("Warehouse", default_warehouse) and not woocommerce_settings.warehouse:
+		frappe.throw(_("Please set Warehouse in Woocommerce Setting"))
+
+	for item in order.get("line_items"):
+		woocomm_item_id = item.get("product_id")
+		found_item = frappe.get_doc("Item", {"woocommerce_id": cstr(woocomm_item_id)})
+
+		ordered_items_tax = item.get("total_tax")
+
+		new_sales_order.append(
+			"items",
+			{
+				"item_code": found_item.name,
+				"item_name": found_item.item_name,
+				"description": found_item.item_name,
+				"delivery_date": new_sales_order.delivery_date,
+				"uom": woocommerce_settings.uom or _("Nos", sys_lang),
+				"qty": item.get("quantity"),
+				"rate": item.get("price"),
+				"warehouse": woocommerce_settings.warehouse or default_warehouse,
+			},
+		)
+
+		add_tax_details(
+			new_sales_order, ordered_items_tax, "Ordered Item tax", woocommerce_settings.tax_account
+		)
+
+	# shipping_details = order.get("shipping_lines") # used for detailed order
+
+	add_tax_details(
+		new_sales_order, order.get("shipping_tax"), "Shipping Tax", woocommerce_settings.f_n_f_account
+	)
+	add_tax_details(
+		new_sales_order,
+		order.get("shipping_total"),
```

Comment:
Logically shipping costs should be added before shipping tax.

Reply:
Also IMO we should review tax computation thoroughly. There were several major issues with the current Shopify integration and I think this implementation of woocommerce taxes feels way to simple to be correct.